discord/client.py

```diff
@@ -485,7 +436,7 @@ async def before_identify_hook(self, shard_id: Optional[int], *, initial: bool =
 
     # login state management
 
-    async def login(self, token: str) -> None:
+    async def login(self, token: str, *, connector: Optional[aiohttp.BaseConnector] = None) -> None:
```

Comment:
No, this should be passed at `__init__`.

Again, if this ends up erroring out then that's *fine*, the error is propagated to the user and anyone with sufficient knowledge of asyncio and aiohttp would know how to fix it.